doc/tutorial/read_mapper/read_mapper_step4.cpp

```diff
@@ -66,19 +66,18 @@ void map_reads(std::filesystem::path const & query_path,
 
     for (auto & [query, id, qual] : query_in)
     {
-        auto positions = search(query, index, search_config);
-        for (auto & [idx, pos] : positions)
+        for (auto & [query_idx, text_pos] : search(query, index, search_config))
         {
-            size_t start = pos ? pos - 1 : 0;
-            std::span text_view{std::data(storage.seqs[idx]) + start, query.size() + 1};
+            size_t start = text_pos.second ? text_pos.second - 1 : 0;
+            std::span text_view{std::data(storage.seqs[text_pos.first]) + start, query.size() + 1};
```

Comment:
Because it is not a search result yet. It will be changed to a search result. So it was before already a pair. Second, the first is the index of the sequence and the second is the position of the hit within the sequence. Anyway, that will be changed to a search result.